src/__tests__/useA11yBlockLink.test.js

```diff
@@ -0,0 +1,144 @@
+import React, {useRef} from 'react';
+import {render, configure, fireEvent} from '@testing-library/react';
+import useA11yBlockLink from '../useA11yBlockLink';
+
+configure({testIdAttribute: 'data-test-id'});
+
+const Demo = ({testId, handleMainClick, handleThirdClick}) => {
+  const mainClickRef = useRef();
+  const secondRef = useRef();
+  const thirdRef = useRef();
+
+  const {handleClick: handleA11yClick} = useA11yBlockLink({
+    mainClick: mainClickRef,
+    secondRef,
+    thirdRef,
+  });
+
+  const handleClick = () => {
+    handleMainClick();
+    handleA11yClick();
+  };
+
+  return (
+    <div onClick={handleMainClick} data-test-id={`${testId}-wrapper`}>
+      <a is="a" ref={mainClickRef} data-test-id={`${testId}-main`} href="#main">
+        <img
+          data-test-id={`${testId}-image`}
+          src="https://dummyimage.com/300x300/000/fff"
+        />
+      </a>
+      <a href="#second" ref={secondRef} data-test-id={`${testId}-second`}>
+        Change Page
+      </a>
+      <p>Description text</p>
+      <button
+        type="button"
+        onClick={handleThirdClick}
+        ref={thirdRef}
+        data-test-id={`${testId}-third`}
+      >
+        Third Event
+      </button>
+    </div>
+  );
+};
+
+describe('useA11yBlockLink', () => {
+  global.window = Object.create(window);
```

Comment:
🤔  Whats going on here? Good opportunity for a comment, same for the lines below?

Reply:
Instead of creating a new global window I updated to using global location and updating. Will add comments.